alacritty_terminal/src/input.rs

```diff
@@ -304,6 +304,12 @@ pub enum Action {
 
     /// No action.
     None,
+
+    /// Mark binding as unset.
+    ///
+    /// It is just a meta action that will be never received. Bindings with this action will be
+    /// removed shortly after deserialization.
+    Unset,
```

Comment:
The more I think about this the more I dislike adding a new "meta"-action. For a few reasons:

1. A new `Unset` meta-action cannot (and should not) ever be executed, making it weird and special
2. The current `None` action's utility remains unknown to me
3. We're trying to communicate something which isn't communicated in the distinction between `None` and `Unset` (at least to me)

Given the historic documentation's wording around `None`, and the lack of a proper use case for a NOOP, I think I'd simply argue for `None` to be re-purposed.

However, this doesn't completely address my first point, namely that we're providing an action which should never be called. I'd like to avoid needing two data structures for this (`ConfigAction` and `Action`), so how might we be able to do this? Allowing a config entry like `{ key: X mods: Shift }` without an action altogether might be an option. Or allow passing an explicit "intrinsic" action, like how "default" here means the baseline (intrinsic) action of `receive_char`, plus any default actions. Something like `{ key: X, mods: Shift, action: ReceiveChar }`. This is awkward, because it's unpredictable. User's aren't going to guess to call a `ReceiveChar` action, although, neither will they guess to call an `Unset` action.

Long story short... can we avoid this meta-action. I'd be a bit happier if we could.

Reply:
> I think I'd simply argue for None to be re-purposed

I think repurposing an existing thing and making it do something completely different is about the worst approach here.

We can't just completely break things without any warnings. There's a good chance there are people out there relying on this and it would be extremely annoying to just make something do something entirely different from one day to another.

Renaming this binding to something different at least gives us the option to deprecate one in the future without breaking user configurations today. I see no reason why we would so actively aim to break things when there's a clear and simple way to resolve this without any breakage.

> I view action names in the config as the name of the behavior I get when I activate the binding from input to Alacritty, not as actions being preformed on the configuration itself.

I agree with this and have never been a huge fan of the different choices for actions proposed here, though I do not have a better proposal myself.

> Long story short... can we avoid this meta-action? I'd be a bit happier if we could.

Fundamentally I agree with this. The question is just how to do it.

As @zsugabubus has already correctly realized, the main problem here might not be how to correctly unbind things, but rather that we cannot have things in the default configuration as the only source of truth because we need to support multiple platforms.

The proposal to have a global `platform` key is also quite troublesome, because it would cause quite a mess, however I think it could be reasonable to have a `platform` key for keybindings (maybe @zsugabubus even meant that in his original comment?). That way we could have all bindings that are questionable and might need to be overwritten (the action bindings) in the default config.

I don't think there ever is any good reason to overwrite the `char` bindings that are used to sent escape sequences. But if someone has a different experience here I'd be very curious. And even then mapping to `chars: ""` should still be possible to overwrite the defaults.

Reply:
To be honest, I also don't like this meta action thing for the same reasons as you @nixpulvis.

If we want a nicer solution, I think, we can close out every possibilities when the user has to specify one-by-one which bindings she wants to unset. Alternatives:
* Introduce config option to disable builtin mappings -> adds unnecessary complexity + not cross-platform friendly.
* Automagically disable builtin mappings based on some kind of heuristics -> inconvenient when someone just wants to extend her bindings and want to keep others + it's too magical + still not cross-platform friendly.
* Make user responsible to maintain her cross-platform config, by placing every platform-specific bindings into user's config (see my previous comments why I consider it useful). I'm sure whoever would use Alacritty on multiple platforms has a complete solution how to maintain different configs across different platforms, since probably not Alacritty would be the only one that needs it. However it's likely more than inconvenient because only these few keybindings are platform-dependent.
* Alacritty handles cross-platform bindings, for example by a `platform` key  (per binding). It's like the previous, except users don't have to mess with different config files. Seems the simplest solution.

I'd vote for the last, but I don't have other just Linux systems, so I would happy with the last two.

Other ideas?

Reply:
I'm against forcing users to add **common** platform bindings to make things work as expected. e.g. I want to run alacritty without default configuration file and everything should **work**, if it'll force me to create config and place some bindings there I'll be pretty angry, since you'll broke my contributor workflow. ( I run `alacritty --config-file /dev/null` a lot to test things).

Reply:
That's where system-wide config file comes in, shipped with Alacritty.

Reply:
Oh yeah that's a great point I completely forgot about myself again. I personally run a completely empty configuration file and I wouldn't want to need part of the default bindings to be required in the configuration file to have them active. That's just unnecessary complexity we're putting in the user config.

We also can't make it that completely empty bindings are different than bindings that are slightly filled, since that makes it extremely hard for the user to understand what's going on.

So I don't immediately see a solution for this which doesn't include having a binding which can override defaults. Though I'd love to hear any suggestions that take this into account.

Reply:
1) You wouldn't have to create config file since if user-local doesn't exists Alacritty falls back to the system-wide.
2) If you would like to test font resizing than you could use a dummy config file with 5 lines. If you don't like it for some reason, then Alacritty could fall back to the system-wide `key_bindings` in such cases.

Reply:
if user config doesn't exist alacritty generates one for it (this behavior will be removed in a near future). 

If you force `alacritty` to use empty config it'll read it and fallback to default options and not to system wide config. alacritty doesn't have a notion of system wide configuration atm.

Reply:
In my opinion `Unset` is the least evil we have here. Keeping some file to make alacritty work is not an option (system-wide/user one, doesn't matter).

Reply:
> if user config doesn't exist alacritty generates one for it (this behavior will be removed in a near future)
> doesn't have a notion of system wide configuration atm

Cannot change these now?

Reply:
Oh, and just in case we introduce some key to distinguish between platforms it won't solve the problem. It'll make alacritty.yml more portable if user decide to add more keybindings for each platform, but you'll need `Unset` action, since generating/relaying on external file is not an option.

Reply:
I don't get what you mean. Why would we need `Unset`?

> since generating/

Alacritty would have to do generate nothing. System-wide config file written once by the "installer", and user-local config only written by user.

> relaying on external file is not an option

We are talking about the config file, don't? Why we can't rely on config files as the source of configuration settings? Or what do you mean?

Reply:
###### Why I see messages only later?

> Keeping some file to make alacritty work is not an option

1) Don't have to. User-local > System-wide > Hard-coded. 👍 
2) Anyway, if you delete Alacritty binary you don't have to be surprised if Alacritty doesn't start up next time. Why keep it? Alacritty comes with sensible defaults that are shipped with the config file.

Reply:
> Alacritty would have to do generate nothing. System-wide config file written once by the "installer", and user-local config only written by user.

We can't create a system-wide config file, that is something distro package maintainers should do. We shouldn't rely on them here and I personally don't want to see `/etc/alacritty/alacritty.yml` file at some point, since it doesn't make sense to me.

> I don't get what you mean. Why would we need Unset?

I personally don't need it, since I'm fine with the current `None` action. However if we want to achieve the initial behavior proposed in your PR we need `Unset` thing ( or something similar ). 

>  Anyway, if you delete Alacritty binary you don't have to be surprised if Alacritty doesn't start up next time. Why keep it? Alacritty comes with sensible defaults that are shipped with the config file.

Why it shouldn't startup, I'm excepting application to work. That's why we have message bar to notify users of broken config, instead of crashing. If alacritty crashes on startup it's a bug, that should be fixed.

> Don't have to. User-local > System-wide > Hard-coded. +1

We already have a relationship User-local > Hard-coded, why do we need system-wide then? The problem is that None do nothing instead of sending chars.

Reply:
> System-wide config file written once by the "installer"

This is not an option. We cannot rely on system-wide configuration files. We need to be able to run on a system without any Alacritty configuration file whatsoever and it should still provide all the default key bindings. There's no way around this.

Reply:
Okay. Made modifications on another branch. See my comment below.